clients/java/src/main/java/io/camunda/zeebe/client/impl/search/response/OperationImpl.java

```diff
@@ -30,8 +30,8 @@ public class OperationImpl implements Operation {
   public OperationImpl(final OperationItem item) {
     id = item.getId();
     batchOperationId = item.getBatchOperationId();
-    type = item.getType();
-    state = item.getState();
+    type = item.getType().toString();
+    state = item.getState().toString();
```

Comment:
@ralfpuchert OperationImpl was added by you, however after rebase with main, I got errors below:

```
[ERROR] COMPILATION ERROR :
[ERROR] /clients/java/src/main/java/io/camunda/zeebe/client/impl/search/response/OperationImpl.java:[33,24] incompatible types: io.camunda.zeebe.client.protocol.rest.OperationItem.TypeEnum cannot be converted to java.lang.String
[ERROR] /clients/java/src/main/java/io/camunda/zeebe/client/impl/search/response/OperationImpl.java:[34,26] incompatible types: io.camunda.zeebe.client.protocol.rest.OperationItem.StateEnum cannot be converted to java.lang.String
```

This is because I declared type and state as enums for the Open API model, same as in Operate code:
https://github.com/camunda/camunda/pull/21552/files#diff-a18635d76659a781a6e8d301bd0fed7c77fd6cc5c8cfc57e11c5bbc19c78c9cdR1589

So I had to add conversion using toString(). What is the correct way to handle such enums in client code ? With toString() as I did, or a different way ? Because I could not find any examples, thank you.

Reply:
FYI this will be fixed as part of https://github.com/camunda/camunda/issues/21836